libminifi/include/core/extension/Module.h

```diff
@@ -0,0 +1,71 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "core/logging/Logger.h"
+#include "utils/gsl.h"
+#include "properties/Configure.h"
+#include "Extension.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace core {
+namespace extension {
+
+/**
+ * Represents an initializable component of the agent.
+ */
+class Module {
```

Comment:
How is module different from extension? It looks like something that can contain multiple extensions, but do we actually need to make a distinction?

Reply:
a module is either the executable or some dlopen-ed dynamic library, opening a dynamic library could trigger the transitive loading of other extensions (`http-curl` -> `civet`) so there could be multiple extensions for a single dlopen